chore: add one rust SDK e2e example (#876)
lancedb/lancedb: docs/src/basic.md

    #[tokio::main]
    async fn main() -> Result<()> {
        let uri = "data/sample-lancedb";
        let db = connect(uri).await?;
    }
    ```

>     !!! info "Only the snippet of the Rust example is displayed. For a fully working example, please check [examples/simple.rs](https://github.com/lancedb/lancedb/tree/main/rust/vectordb/src/examples/simple.rs)."

Reviewer: ```suggestion
    !!! info "See [examples/simple.rs](https://github.com/lancedb/lancedb/tree/main/rust/vectordb/src/examples/simple.rs) for a full working example."
```

    ```

By default, LanceDB runs a brute-force scan over dataset to find the K nearest neighbours (KNN).

> If there are more than 50K vectors in one table, it is suggested to create vector index to speed up
> search.

Reviewer: ```suggestion
For tables with more than 50K vectors, creating an ANN index is recommended to speed up search performance.
```
